server/src/main/java/org/candlepin/policy/SystemPurposeComplianceRules.java

```diff
@@ -198,7 +198,12 @@ public SystemPurposeComplianceStatus getStatus(Consumer consumer,
                 if (StringUtils.isNotEmpty(preferredSla) &&
                     product.hasAttribute(Product.Attributes.SUPPORT_LEVEL)) {
                     String sla = product.getAttributeValue(Product.Attributes.SUPPORT_LEVEL);
-                    if (sla.equalsIgnoreCase(preferredSla)) {
+                    //TODO: This is a short term fix for bug 1858167. Added a condition to match
+                    // Syspurpose status for entitlement which has SLA as Layered and SLA exempt as true
+                    boolean slaExempt = Boolean.parseBoolean(product
+                        .getAttributeValue(Product.Attributes.SUPPORT_LEVEL_EXEMPT));
+                    if ((sla.equalsIgnoreCase("Layered") && slaExempt) ||
```

Comment:
If we are indeed going to apply this solution, then we should only check the `slaExempt` attribute, and not if the sla is `Layered`. From what I understand, currently only Layered SLAs are considered exempt for now, but you never know if in the future we might have another SLA type that is also considered exempt.

Reply:
I agree with Nikos.   There should be no hard-coded reference to 'Layered'; slaExempt is what is imprtant.

Reply:
Agreed,  updated the condition to check only `slaExempt`.